changes.go

```diff
@@ -0,0 +1,207 @@
+package apidApigeeSync
+
+import (
+	"github.com/30x/apid-core"
+	"time"
+	"net/url"
+	"path"
+	"net/http"
+	"io/ioutil"
+	"encoding/json"
+
+	"github.com/apigee-labs/transicator/common"
+
+)
+
+var lastSequence string
+var block        string = "45"
+
+/*
+ * Long polls the change agent with a 45 second block. Parses the response from
+ * change agent and raises an event. Called by pollWithRetry().
+ */
+func pollChangeAgent(quit chan bool) error {
+
+	changesUri, err := url.Parse(config.GetString(configChangeServerBaseURI))
+	if err != nil {
+		log.Errorf("bad url value for config %s: %s", changesUri, err)
+		return err
+	}
+	changesUri.Path = path.Join(changesUri.Path, "changes")
+
+	/*
+	 * Check to see if we have lastSequence already saved in the DB,
+	 * in which case, it has to be used to prevent re-reading same data
+	 */
+	lastSequence = getLastSequence()
+
+	for {
+		select {
```

Comment:
The behavior of this select is:
If `quit` is not available, it goes into `default`. Since `getChanges ` is blocking (using http), when receiving `quit`, this func will not return immediately if it is blocked in `getChanges`. It will not quit until it finishes `getChanges `, in the worst case, after the timeout `httpTimeout + httpTimeout`.
I'm not sure what is your expected behavior, but if you want the `quit` to take effect immediately, I would call `getChanges` in another go routine like this:
```
done := make(chan error, 1)
go getChanges(changesUri, done)
for {
  select {
  case <- quit:
    return quitSignalError{}
  case err := <- done:
    return err
  }
}

```